pkg/otlp/attributes/source.go

```diff
@@ -48,15 +51,17 @@ func getClusterName(attrs pcommon.Map) (string, bool) {
 
 // hostnameFromAttributes tries to get a valid hostname from attributes by checking, in order:
 //
-//  1. a custom Datadog hostname provided by the "datadog.host.name" attribute
+//  1. the "host" attribute to avoid double tagging if present.
 //
-//  2. cloud provider specific hostname for AWS, Azure or GCP,
+//  2. a custom Datadog hostname provided by the "datadog.host.name" attribute
 //
-//  3. the Kubernetes node name (and cluster name if available),
+//  3. cloud provider specific hostname for AWS, Azure or GCP,
 //
-//  4. the cloud provider host ID and
+//  4. the Kubernetes node name (and cluster name if available),
 //
-//  5. the host.name attribute.
+//  5. the cloud provider host ID and
+//
+//  6. the host.name attribute.
```

Comment:
This is just adding the top line and bumping the numbers on the rest of items, but the resulting diff is not very readable